sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodOverloadOptionalParameter.cs

```diff
@@ -147,18 +148,28 @@ private static bool IsMethodHidingOriginal(IMethodSymbol candidateHidingMethod,
         {
             for (int i = 0; i < candidateHidingMethod.Parameters.Length; i++)
             {
-                if (!candidateHidingMethod.Parameters[i].Type.Equals(method.Parameters[i].Type))
+                var candidateParam = candidateHidingMethod.Parameters[i];
+                var methodParam = method.Parameters[i];
+
+                if (!Equals(candidateParam.Type, methodParam.Type) &&
+                    !AreTypeParameters(candidateParam, methodParam) &&
+                    !Equals(candidateParam.Type.OriginalDefinition, methodParam.Type.OriginalDefinition))
                 {
                     return false;
                 }
 
-                if (candidateHidingMethod.Parameters[i].IsOptional != method.Parameters[i].IsOptional)
+                if (candidateParam.IsOptional != methodParam.IsOptional)

```

Comment:
I considered it as well, but decided against. Mainly because this would result in expression with three ANDs and one OR, which looks a bit ugly and a bit confusing.